txpool/txpool_test.go

```diff
@@ -1,1125 +1,1232 @@
+				GasPrice: big.NewInt(1),
+				Gas:      validGasLimit,
+			},
+		},
+		{
+			name:          "ErrInvalidSender",
+			expectedError: ErrInvalidSender,
+			tx: &types.Transaction{
+				From:     types.ZeroAddress,
+				Value:    big.NewInt(1),
+				GasPrice: big.NewInt(1),
+				Gas:      validGasLimit,
+			},
+		},
+		{
+			name:          "ErrInvalidAccountState",
+			expectedError: ErrInvalidAccountState,
+			tx: &types.Transaction{
+				From:     types.Address{0x1},
+				Value:    big.NewInt(1),
+				GasPrice: big.NewInt(1),
+				Gas:      validGasLimit,
+			},
+		},
+		{
+			name:          "ErrTxPoolOverflow",
+			expectedError: ErrTxPoolOverflow,
+			tx: &types.Transaction{
+				From:     types.Address{0x1},
+				Value:    big.NewInt(1),
+				GasPrice: big.NewInt(1),
+				Gas:      validGasLimit,
+			},
+		},
+		{
+			name:          "ErrIntrinsicGas",
+			expectedError: ErrIntrinsicGas,
+			tx: &types.Transaction{
+				From:     types.Address{0x1},
+				Value:    big.NewInt(1),
+				GasPrice: big.NewInt(1),
+				Gas:      1,
+			},
+		},
+		{
+			name:          "ErrAlreadyKnown",
+			expectedError: ErrAlreadyKnown,
+			tx: &types.Transaction{
+				From:     types.Address{0x1},
+				Value:    big.NewInt(1),
+				GasPrice: big.NewInt(1),
+				Gas:      validGasLimit,
+			},
 		},
 		{
-			name:          "should fail with gas too low error",
-			txValue:       big.NewInt(10),
-			txGasLimit:    1,
-			txGasPrice:    big.NewInt(1),
-			shouldSucceed: false,
+			name:          "ErrOversizedData",
+			expectedError: ErrOversizedData,
+			tx: &types.Transaction{
+				From:     types.Address{0x1},
+				Value:    big.NewInt(1),
+				GasPrice: big.NewInt(1),
+				Gas:      validGasLimit,
+			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			pool, err := NewTxPool(hclog.NewNullLogger(), false, nil, false, defaultPriceLimit, defaultMaxSlots, forks.At(0), &mockStore{}, nil, nil, nilMetrics)
-			if err != nil {
-				t.Fatal("Failed to initialize transaction pool:", err)
-			}
-			signer := crypto.NewEIP155Signer(100)
-			pool.AddSigner(signer)
-
-			tx := &types.Transaction{
-				To:       &receiverAddr,
-				Value:    tc.txValue,
-				Gas:      tc.txGasLimit,
-				GasPrice: tc.txGasPrice,
+			pool, err := newTestPool()
+			assert.NoError(t, err)
+			pool.EnableDev()
+			pool.AddSigner(crypto.NewEIP155Signer(100))
+
+			/* special error setups */
+			switch tc.expectedError {
+			case ErrTxPoolOverflow:
+				pool.gauge.increase(defaultMaxSlots)
+			case ErrNonEncryptedTx:
+				pool.dev = false
+			case ErrAlreadyKnown:
+				go func() {
+					err := pool.addTx(local, tc.tx)
+					assert.NoError(t, err)
+				}()
+				go pool.handleAddRequest(<-pool.addReqCh)
+				<-pool.promoteReqCh
+			case ErrInvalidAccountState:
+				pool.store = faultyMockStore{}
+			case ErrOversizedData:
+				data := make([]byte, 989898)
+				_, err := rand.Read(data)
+				assert.NoError(t, err)
+				tc.tx.Input = data
```

Comment:
I agree that conditional branching in unit tests is not recommended, but with the `t.Run()` approach, the test would be 10 times longer with identical logic in each case except for a few lines of code. This switch case is just meant to avoid that. Everything that's common to these cases is already achieved with `newTestPool`.